geodata/models/raster/reader.py

```diff
@@ -0,0 +1,128 @@
+"""Helper methods for creating a ``GDALRaster`` entry from a raster file."""
+from celery.utils.log import get_task_logger
+from django.contrib.gis.gdal import SpatialReference
+from django.contrib.gis.geos import Polygon
+import rasterio
+
+# from django.core.files.base import ContentFile
+# import io
+# from PIL import Image
+
+from .base import BandMetaEntry, ConvertedRasterFile, RasterEntry, RasterFile
+from ..common import _ReaderRoutine
+
+logger = get_task_logger(__name__)
+
+
+class RasterEntryReader(_ReaderRoutine):
+    """Raster injestion routine.
+
+    This helper class will open a raster file and create the ``GDALRaster``
+    for the ``RasterEntry``'s raster property.
+
+    """
+
+    # @staticmethod
+    # def create_thumbnail(src):
+    #     oview = int(max(src.height, src.width) * 0.1) or 1
+    #     thumbnail = src.read(1, out_shape=(1, int(src.height // oview), int(src.width // oview)))
+    #
+    #     buf = io.BytesIO()
+    #     thumbnail.save(buf, format='JPEG')
+    #     byte_im = buf.getvalue()
+    #     return ContentFile(byte_im)
+
+    def _read_files(self):
+        """Load raster data files and create a ``RasterEntry``.
+
+        This function will need to handle a lot of scenarios for how
+        the raster file is passed and where it is stored.
+
+        """
+        # Fetch the raster file this Layer corresponds to
+        self.rfe = RasterFile.objects.get(id=self.model_id)
+        file_path = self.rfe.raster_file.name
+
+        logger.info(f'The raster file path: {file_path}')
+
+        raster_querry = RasterEntry.objects.filter(raster_file=self.rfe)
+        if len(raster_querry) < 1:
+            self.raster_entry = RasterEntry()
```

Comment:
When we create a new RasterEntry, we should set the name to the raster file's name be default.